src/Application/Messenger/Handler/PersonHandler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace MatchBot\Application\Messenger\Handler;
+
+use MatchBot\Domain\DonorAccount;
+use MatchBot\Domain\DonorAccountRepository;
+use MatchBot\Domain\StripeCustomerId;
+use Messages\Person;
+use Psr\Container\ContainerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+
+/**
+ * Saves new or updated Person info. Expected only for those with passwords and Stripe Customer
+ * records.
+ */
+#[AsMessageHandler]
+readonly class PersonHandler
+{
+    /** @psalm-suppress PossiblyUnusedMethod Used by Messenger mapping & tests. */
+    public function __construct(
+        // apparently at the time this is constructed in tests the container isn't ready to give
+        // us a donorAccountRepository, so taking a ref to the container instead and getting the
+        // repository inside __invoke
+        private ContainerInterface $container,
```

Comment:
Strategy & comment borrowed from charity update counterpart. I found that when I injected the repo itself, actual runtime worked but tests hit strange errors from repo's not being instantiable in DI and ending up null.